apps/api/src/user/user.service.ts

```diff
@@ -215,6 +215,25 @@ export class UserService {
     });
   }
 
+  async upsertUserPreference(userId: string, preferenceData: Partial<UserPreference>) {
+    const user = await this.findOne(userId);
+
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    if (!user.userPreference) {
+      const userPreference = await this.userPreferenceRepo.save(
+        this.userPreferenceRepo.create(preferenceData),
+      );
+
+      user.userPreference = userPreference;
+      await this.userRepo.save(user);
+    }
+
+    return user.userPreference;
+  }
+
```

Comment:
This method would work well in today's scenario when we do not have a saved userpreference for any user. However, if the userpreference already existed for a user, and you were adding a new pref to an existing one? This wouldn't.